File: BikeDemoJava/app/src/main/java/com/magiclane/sdk/examples/bikedemojava/SettingsSliderItem.java
/*
 * SPDX-FileCopyrightText: 2026 Magic Lane International B.V. <info@magiclane.com>
 * SPDX-License-Identifier: Apache-2.0
 *
 * Contact Magic Lane at <info@magiclane.com> for SDK licensing options.
 */

package com.magiclane.sdk.examples.bikedemojava;

public class SettingsSliderItem extends SettingsItem {
    private final float valueFrom;
    private final float value;
    private final float valueTo;
    private final String unit;
    private final SliderCallback callback;

    public interface SliderCallback {
        void onChanged(float value);
    }

    public SettingsSliderItem(String title, float valueFrom, float value, float valueTo, String unit, SliderCallback callback) {
        super(title);
        this.valueFrom = valueFrom;
        this.value = value;
        this.valueTo = valueTo;
        this.unit = unit;
        this.callback = callback;
    }

    public float getValueFrom() {
        return valueFrom;
    }

    public float getValue() {
        return value;
    }

    public float getValueTo() {
        return valueTo;
    }

    public String getUnit() {
        return unit;
    }

    public SliderCallback getCallback() {
        return callback;
    }
}

File: BikeSimulation/app/src/main/kotlin/com/magiclane/sdk/examples/bikesimulation/SettingsSliderItem.kt
// -------------------------------------------------------------------------------------------------

/*
 * Copyright (C) 2019-2024, Magic Lane B.V.
 * All rights reserved.
 *
 * This software is confidential and proprietary information of Magic Lane
 * ("Confidential Information"). You shall not disclose such Confidential
 * Information and shall use it only in accordance with the terms of the
 * license agreement you entered into with Magic Lane.
 */

// -------------------------------------------------------------------------------------------------

package com.magiclane.sdk.examples.bikesimulation

data class SettingsSliderItem(
    override val title: String = "",
    val valueFrom: Float = 0f,
    var value: Float = 0f,
    val valueTo: Float = 0f,
    val unit: String = "",
    val callback: (Float) -> Unit
) : SettingsItem(title)
